app/javascript/components/volunteer/EditProfileForm.jsx

```diff
@@ -0,0 +1,290 @@
+    let { last_nameValid } = this.state;
+    let { emailValid } = this.state;
+    let { passwordValid } = this.state;
+    let { passwordMatch } = this.state;
+    switch (fieldName) {
+      case "first_name":
+        first_nameValid = value.length > 0;
+        formErrors.first_name = first_nameValid
+          ? ""
+          : " is not a valid first name";
+        break;
+      case "last_name":
+        last_nameValid = value.length > 0;
+        formErrors.last_name = last_nameValid ? "" : " is not a valid last name";
+        break;
+      case "email":
+        emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
+        formErrors.email = emailValid ? "" : " is an invalid email";
+        break;
+      case "password":
+        passwordValid = value.length >= 6;
+        formErrors.password = passwordValid ? "" : " is too short";
+        break;
+      case "password_confirmation":
+        passwordMatch = value.match(this.state.password);
+        formErrors.password = passwordMatch ? "" : " does not match";
+        break;
+      default:
+        break;
+    }
+    this.setState(
+      {
+        formErrors,
+        first_nameValid,
+        last_nameValid,
+        emailValid,
+        passwordValid,
+        passwordMatch
+      },
+      this.validateForm
+    );
+  };
+
+  handleChange = name => {
+    if (!this.handlers[name]) {
+      this.handlers[name] = event => {
+        const { project } = this.state;
+        project[name] = event.target.value;
+        this.setState({ project });
+      };
+    }
+    return this.handlers[name];
+  };
+
+  handleFileChange = e => {
+    this.setState({ selected_file: e.target.files[0] });
+  };
+
+  handleResumeFileChange = e => {
+    this.setState({ selected_resume_file: e.target.files[0] });
+  };
+
+
+  handleChange = name => event => {
+    const { value } = event.target;
+    this.setState({ [name]: value }, () => {
+      this.validateField(name, value);
+    });
+  };
+
+  goBack = e => {
+    e.preventDefault();
+    window.location.href = "/";
+  };
+
+  handleSubmit() {
+    const formData = new FormData();
+    formData.append("user[email]", this.state.email);
+    formData.append("user[password]", this.state.password);
+    formData.append(
+      "user[password_confirmation]",
+      this.state.password_confirmation
+    );
+    formData.append(
+      "user[first_name]",
+      this.state.first_name
+    );
+    formData.append(
+      "user[last_name]",
+      this.state.last_name
+    );
+    formData.append("user[city]", this.state.city);
+    formData.append("user[state]", this.state.state);
+    formData.append("user[link]", this.state.linkedIn);
+    formData.append("user[description]", this.state.bio);
+    formData.append(
+      "user[phone_number]",
+      this.state.phone_number
+    );
+    formData.append("user[profile_image]", this.state.selected_file);
+    axios
+      .put("/users/" + {this.state.project.id}, formData)
```

Comment:
I think you have to change this to `.put("/users/" + this.state.project.id, formData)` for it to work. It errors out for me otherwise.

Reply:
Actually, this should be `this.props.uesr.id`, since you want the user id.